### PR TITLE
chore: update code and docs to fit golang standard

### PR DESCRIPTION
**Pull request by bjwswang**
Hi @tazarov  This pr made some tiny changes without breaking any functinalities:
1. bump golang from go1.18 go go1.20
2.  rename `main.go` to `chroma.go` since `main.go` shall be used as a main entrypoint a runable program
3. use `io` instead of `io/ioutil` due to `io/ioutil` has been deprecated
4. remove some models(not used) in `go.mod`
5.  update `README` to fix type issue in the example

### CODE REVIEW

**Comment by bjwswang** [comment]
@tazarov One more quesion,why do we use `float32` instead of `float64` ?

**Comment by tazarov** [comment]
@bjwswang, just couple of comments above, if you can review?

**Comment by bjwswang** [comment]
> @bjwswang, just couple of comments above, if you can review?

Sure. Please.

**Comment by bjwswang** [comment]
@tazarov  I don't see any comments. is there something wrong?

**Comment by bjwswang** [comment]
@tazarov  The reason I bump go to 1.20 can be found at this annoucement
https://tip.golang.org/doc/go1.20 
1.20 made some optimizations and new features to golang. And one more important reason is that [langchaingo](https://github.com/tmc/langchaingo/blob/main/go.mod) use `1.20`  ^_^ It's better we keep up with the go community.

**Review by tazarov — COMMENTED** [review]
Have a look at the couple of comments I left.

**Comment by tazarov** [comment]
> @tazarov The reason I bump go to 1.20 can be found at this annoucement
> https://tip.golang.org/doc/go1.20
> 1.20 made some optimizations and new features to golang. And one more important reason is that [langchaingo](https://github.com/tmc/langchaingo/blob/main/go.mod) use `1.20` ^_^ It's better we keep up with the go community.

I understand, but is there a direct dependency between projects using different Go versions? Aren't things backwards compatible? What if some users cannot upgrade their golang versions (e.g. running in private/corporate clouds)?

File: README.md [hunk 1]
```diff
@@ -49,14 +49,12 @@ import (
 
 ## Usage
 
-
 Ensure you have a running instance of Chroma running. We recommend one of the two following options:
 
-- Official documentation - https://docs.trychroma.com/usage-guide#running-chroma-in-clientserver-mode
-- If you are a fan of Kubernetes, you can use the Helm chart - https://github.com/amikos-tech/chromadb-chart (Note: You
+- Official documentation - <https://docs.trychroma.com/usage-guide#running-chroma-in-clientserver-mode>
```
[the remark this answers is not in the document]
Comment (bjwswang) [reply README.md:54]: That's auto-completed by my vscode editor.Or we embed the url like  `[Official documentation](https://docs.trychroma.com/usage-guide#running-chroma-in-clientserver-mode)`

File: go.mod [hunk 2]
```diff
@@ -1,19 +1,14 @@
 module github.com/amikos-tech/chroma-go
 
-go 1.18
```
[the remark this answers is not in the document]
Comment (bjwswang) [reply go.mod:3]: @tazarov The reason I bump go to 1.20 can be found at this annoucement
https://tip.golang.org/doc/go1.20
1.20 made some optimizations and new features to golang. 
One more important reason is that [langchaingo](https://github.com/tmc/langchaingo/blob/main/go.mod) use 1.20 ^_^ It's better we keep up with the go community.
Besides kubernetes also utilizes `1.20`
https://github.com/kubernetes/kubernetes/blob/master/go.mod#L9C6-L9C6

File: swagger/client.go
[the remark this answers is not in the document]
Comment (bjwswang) [reply swagger/client.go:1]: How did these files got generated by the way?

[the remark this answers is not in the document]
Comment (bjwswang) [reply swagger/client.go:1]: @tazarov  You are right. Reverted.

**Comment by bjwswang** [comment]
> > @tazarov The reason I bump go to 1.20 can be found at this annoucement
> > https://tip.golang.org/doc/go1.20
> > 1.20 made some optimizations and new features to golang. And one more important reason is that [langchaingo](https://github.com/tmc/langchaingo/blob/main/go.mod) use `1.20` ^_^ It's better we keep up with the go community.
> 
> I understand, but is there a direct dependency between projects using different Go versions? Aren't things backwards compatible? What if some users cannot upgrade their golang versions (e.g. running in private/corporate clouds)?

Well... that depends on the purpose and the people who use this chroma-go.  From my point of view, I am trying to use chroma-go along with langchaingo which have almost all I need to develop AI applications in Golang.  I believe many gophers      are in the same situation as me.

**Comment by bjwswang** [comment]
> 

That's fine if chroma-go keeps using 1.18 untils I found any `must-upgrade` reasons.

File: go.mod (hunk 2 above)
[the remark this answers is not in the document]
Comment (tazarov) [reply go.mod]: @bjwswang, as discussed. Let's keep at 1.18 until we see breaking changes requiring an upgrade.

File: go.mod [hunk 3]
```diff
@@ -5,15 +5,10 @@ go 1.18
 require (
 	github.com/joho/godotenv v1.5.1
 	github.com/stretchr/testify v1.8.4
-	golang.org/x/oauth2 v0.10.0
```
Comment (bjwswang) [inline go.mod:8]: For theses unused packages,I believe this comes from gen_api.sh

File: go.mod (hunk 2 above)
[the remark this answers is not in the document]
Comment (bjwswang) [reply go.mod]: Done @tazarov

File: go.mod (hunk 3 above)
Comment (tazarov) [reply go.mod:8]: It might be the case, yeah. But since this is run locally, they got accidentally committed.

**Comment by bjwswang** [comment]
@tazarov  Seems workflow failed due to `panic: API Key not set`

**Comment by bjwswang** [comment]
> @tazarov Seems workflow failed due to `panic: API Key not set`

I think I know the reason... Actions triggered by Pull request do no have the permission to read secrets

**Comment by tazarov** [comment]
Yeah, I noticed. Let me move EF testing to a separate WF so PRs are not all that affected.

**Comment by bjwswang** [comment]
> Yeah, I noticed. Let me move EF testing to a separate WF so PRs are not all that affected.

Please be aware that I add another job in `go.yml` to enable lint tests.

File: chroma.go [hunk 4]
```diff
@@ -338,7 +344,7 @@ func getMetadatasListFromAPI(metadatas []map[string]openapiclient.MetadatasInner
 			var rawValue interface{}
 			b, e := value.MarshalJSON()
 			if e != nil {
-				rawValue = nil
+				return nil, e
```
Comment (bjwswang) [inline chroma.go:347]: @tazarov golangci-lint reports a error at this place.  I think we should return this error if not nil

File: .github/workflows/go.yml [hunk 5]
```diff
@@ -13,24 +13,23 @@ jobs:
 
   build:
     runs-on: ubuntu-latest
+    environment: Test
```
Comment (bjwswang) [inline .github/workflows/go.yml:16]: @tazarov  If you wanto to use secrets in this Pull request, you can 
1. create a new environment in your github acitons  (environment `Test` )
image
2. create secrets inner that environment ( environment `Test` for example)
3.  set reviewers for this environment 
image

4. use this environment in your workflow just like this `build` job

Comment (bjwswang) [reply .github/workflows/go.yml:16]: When you trigger the workflow, this job need another manual approval so it can read secrets in that environment.

**Review by bjwswang — COMMENTED** [review]
Please see my comments on the workflow changes.

**Comment by bjwswang** [comment]
Hey @tazarov ! Kindly question hre. Are you still focusing this porject?   ^_^

**Comment by tazarov** [comment]
Hey @bjwswang, yes, in my spare time. Thanks for your patience. I will get this merged today.

**Comment by bjwswang** [comment]
> Hey @bjwswang, yes, in my spare time. Thanks for your patience. I will get this merged today.

That's cool @tazarov !

**Pull request merged**